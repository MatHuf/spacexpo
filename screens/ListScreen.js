import React from "react";
import { StyleSheet, Text, SafeAreaView, FlatList } from "react-native";

import { useQuery } from "@apollo/react-hooks";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  },
  title: {
    fontSize: 32
  }
});

export default function ListScreen(props) {
  const { navigation } = props;

  // TODO figure out how to work with Jest and this hook
  const { loading, data } = useQuery(navigation.getParam("query"));

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.title}>{navigation.getParam("title")}</Text>
      {loading ? (
        <Text style={styles.title}>Loading ...</Text>
      ) : (
        data && (
          <FlatList
            data={data[navigation.getParam("listKey")]}
            renderItem={navigation.getParam("renderItem")}
            keyExtractor={item => `${item.id}`}
          />
        )
      )}
    </SafeAreaView>
  );
}
