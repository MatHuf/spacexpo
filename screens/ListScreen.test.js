import React from "react";
import renderer from "react-test-renderer";
import ListScreen from "./ListScreen";
import NavigationMock from "../mocks/navigationMock";
import {
  launchesNavProps,
  historyNavProps,
  capsulesNavProps,
  rocketsNavProps,
  landingPadsNavProps
} from "../navigation/navigationProps";

const launchesNavMock = new NavigationMock(launchesNavProps);
const historyNavMock = new NavigationMock(historyNavProps);

describe("ListScreen: Launches", () => {
  it("renders correctly", () => {
    const tree = renderer.create(() => ListScreen(launchesNavMock)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("ListScreen: History", () => {
  it("renders correctly", () => {
    const tree = renderer.create(() => ListScreen(historyNavMock)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("ListScreen: Capsules", () => {
  it("renders correctly", () => {
    const tree = renderer.create(() => ListScreen(capsulesNavProps)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("ListScreen: Rockets", () => {
  it("renders correctly", () => {
    const tree = renderer.create(() => ListScreen(rocketsNavProps)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe("ListScreen: Landing Pads", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(() => ListScreen(landingPadsNavProps))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
