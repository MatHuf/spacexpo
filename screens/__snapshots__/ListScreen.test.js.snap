// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListScreen: Capsules renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "crew_capacity": 0,
          "description": "Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS).",
          "first_flight": "2010-12-08",
          "id": "dragon1",
          "name": "Dragon 1",
        },
        Object {
          "crew_capacity": 7,
          "description": "Dragon 2 (also Crew Dragon, Dragon V2, or formerly DragonRider) is the second version of the SpaceX Dragon spacecraft, which will be a human-rated vehicle. It includes a set of four side-mounted thruster pods with two SuperDraco engines each, which can serve as a launch escape system or launch abort system (LAS). In addition, it has much larger windows, new flight computers and avionics, and redesigned solar arrays, and a modified outer mold line from the initial cargo Dragon that has been flying for several years.",
          "first_flight": "2019-03-02",
          "id": "dragon2",
          "name": "Dragon 2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              Dragon 1
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              Crew: 
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              First flight: 
              2010-12-08
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS).
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              Dragon 2
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              Crew: 
              7
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              First flight: 
              2019-03-02
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Dragon 2 (also Crew Dragon, Dragon V2, or formerly DragonRider) is the second version of the SpaceX Dragon spacecraft, which will be a human-rated vehicle. It includes a set of four side-mounted thruster pods with two SuperDraco engines each, which can serve as a launch escape system or launch abort system (LAS). In addition, it has much larger windows, new flight computers and avionics, and redesigned solar arrays, and a modified outer mold line from the initial cargo Dragon that has been flying for several years.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`ListScreen: History renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "details": "Falcon 1 becomes the first privately developed liquid fuel rocket to reach Earth orbit.",
          "event_date_utc": "2008-09-28T23:15:00Z",
          "id": 1,
          "title": "Falcon 1 Makes History",
        },
        Object {
          "details": "NASA awards SpaceX $1.6B Commercial Resupply Services (CRS) contract.",
          "event_date_utc": "2008-12-23T01:00:00Z",
          "id": 2,
          "title": "SpaceX Wins $1.6B CRS Contract",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              Falcon 1 Makes History
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              2008-09-28T23:15:00Z
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Falcon 1 becomes the first privately developed liquid fuel rocket to reach Earth orbit.
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              SpaceX Wins $1.6B CRS Contract
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              2008-12-23T01:00:00Z
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              NASA awards SpaceX $1.6B Commercial Resupply Services (CRS) contract.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`ListScreen: Launches renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "details": "Engine failure at 33 seconds and loss of vehicle",
              "flight_number": 1,
              "launch_date_utc": "2006-03-24T22:30:00.000Z",
              "launch_site": Object {
                "site_name_long": "Kwajalein Atoll Omelek Island",
              },
              "rocket": Object {
                "rocket_name": "Falcon 1",
              },
            },
          ],
          "title": "2006",
        },
        Object {
          "data": Array [
            Object {
              "details": "Successful first stage burn and transition to second stage, maximum altitude 289 km, Premature engine shutdown at T+7 min 30 s, Failed to reach orbit, Failed to recover first stage",
              "flight_number": 2,
              "launch_date_utc": "2007-03-21T01:10:00.000Z",
              "launch_site": Object {
                "site_name_long": "Kwajalein Atoll Omelek Island",
              },
              "rocket": Object {
                "rocket_name": "Falcon 1",
              },
            },
          ],
          "title": "2007",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "details": "Engine failure at 33 seconds and loss of vehicle",
              "flight_number": 1,
              "launch_date_utc": "2006-03-24T22:30:00.000Z",
              "launch_site": Object {
                "site_name_long": "Kwajalein Atoll Omelek Island",
              },
              "rocket": Object {
                "rocket_name": "Falcon 1",
              },
            },
          ],
          "title": "2006",
        },
        Object {
          "data": Array [
            Object {
              "details": "Successful first stage burn and transition to second stage, maximum altitude 289 km, Premature engine shutdown at T+7 min 30 s, Failed to reach orbit, Failed to recover first stage",
              "flight_number": 2,
              "launch_date_utc": "2007-03-21T01:10:00.000Z",
              "launch_site": Object {
                "site_name_long": "Kwajalein Atoll Omelek Island",
              },
              "rocket": Object {
                "rocket_name": "Falcon 1",
              },
            },
          ],
          "title": "2007",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
        3,
      ]
    }
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#777777",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 22,
              }
            }
          >
            2006
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              Launch 
              1
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              Falcon 1
               | 
              Kwajalein Atoll Omelek Island
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Engine failure at 33 seconds and loss of vehicle
              .
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#777777",
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 22,
              }
            }
          >
            2007
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              Launch 
              2
            </Text>
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 18,
                }
              }
            >
              Falcon 1
               | 
              Kwajalein Atoll Omelek Island
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Successful first stage burn and transition to second stage, maximum altitude 289 km, Premature engine shutdown at T+7 min 30 s, Failed to reach orbit, Failed to recover first stage
              .
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`ListScreen: Rockets renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`ListScreen: Rockets renders correctly 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "attempted_landings": 0,
          "ship_id": "ASOG",
          "ship_name": "A Shortfall of Gravitas",
          "successful_landings": 0,
        },
        Object {
          "attempted_landings": 2,
          "ship_id": "JRTI-1",
          "ship_name": "Just Read The Instructions",
          "successful_landings": 0,
        },
        Object {
          "attempted_landings": 8,
          "ship_id": "JRTI-2",
          "ship_name": "Just Read The Instructions",
          "successful_landings": 7,
        },
        Object {
          "attempted_landings": 28,
          "ship_id": "OCISLY",
          "ship_name": "Of Course I Still Love You",
          "successful_landings": 24,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              ASOG
               | 
              A Shortfall of Gravitas
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Successful landings: 
              0
              /
              0
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              JRTI-1
               | 
              Just Read The Instructions
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Successful landings: 
              0
              /
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              JRTI-2
               | 
              Just Read The Instructions
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Successful landings: 
              7
              /
              8
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "x": -5,
                "y": -5,
              },
              "shadowOpacity": 50,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 22,
                }
              }
            >
              OCISLY
               | 
              Of Course I Still Love You
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "paddingTop": 10,
                  "textAlign": "justify",
                }
              }
            >
              Successful landings: 
              24
              /
              28
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
